app/src/main/java/woowacourse/movie/view/extension/ParcelableExtension.kt

```
@@ -0,0 +1,21 @@
+
+inline fun <reified T> Intent.getParcelableCompat(key: String): T =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        this.getParcelableExtra(key, T::class.java) ?: throw IllegalArgumentException()
+    } else {
+        @Suppress("DEPRECATION")
+        getParcelableExtra(key) as? T ?: throw IllegalArgumentException()
+    }
+
+inline fun <reified T> Bundle.getParcelableCompat(key: String): List<T> =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        getParcelableArray(key, T::class.java)?.toList() ?: throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)